src/main/java/org/jenkinsci/plugins/github_branch_source/DefaultGitHubNotificationStrategy.java

```diff
@@ -0,0 +1,124 @@
+            listener.getLogger().println(
+                    "Can not determine Jenkins root URL. Commit status notifications are disabled "
+                            + "until a root URL is"
+                            + " configured in Jenkins global configuration.");
+        }
+        return url;
+    }
+
+    /**
+     * @since TODO
+     */
+    protected String generateMessage(GitHubNotificationContext notificationContext, TaskListener listener) {
+        Run<?, ?> build = notificationContext.getBuild();
+        if (null != build) {
+            Result result = build.getResult();
+            if (Result.SUCCESS.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Good();
+            } else if (Result.UNSTABLE.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Unstable();
+            } else if (Result.FAILURE.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Failure();
+            } else if (Result.ABORTED.equals(result)) {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Aborted();
+            } else if (result != null) { // NOT_BUILT etc.
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Other();
+            } else {
+                return Messages.GitHubBuildStatusNotification_CommitStatus_Pending();
+            }
+        }
+        return Messages.GitHubBuildStatusNotification_CommitStatus_Queued();
+    }
+
+    /**
+     * @since TODO
+     */
+    protected GHCommitState generateState(GitHubNotificationContext notificationContext, TaskListener listener) {
+        Run<?, ?> build = notificationContext.getBuild();
+        if (null != build) {
+            Result result = build.getResult();
+            if (Result.SUCCESS.equals(result)) {
+                return GHCommitState.SUCCESS;
+            } else if (Result.UNSTABLE.equals(result)) {
+                return GHCommitState.FAILURE;
+            } else if (Result.FAILURE.equals(result)) {
+                return GHCommitState.ERROR;
+            } else if (Result.ABORTED.equals(result)) {
+                return GHCommitState.ERROR;
+            } else if (result != null) { // NOT_BUILT etc.
+                return GHCommitState.ERROR;
+            }
+        }
+        return GHCommitState.PENDING;
+    }
+
+    /**
+     * @since TODO
+     */
+    protected boolean ignoreError(GitHubNotificationContext notificationContext, TaskListener listener) {
+        Run<?, ?> build = notificationContext.getBuild();
+        return null == build || null == build.getResult();
+    }
+
+    /**
+     * @since TODO
+     */
+    public List<GitHubNotificationRequest> notifications(GitHubNotificationContext notificationContext, TaskListener listener) {
+        return new ArrayList<>(Arrays.asList(new GitHubNotificationRequest(generateContext(notificationContext, listener),
```

Comment:
No need to wrap in an ArrayList when you are using `Arrays.asList` already... and as you are only returning one that is probably better to use `Collections.singletonList` instead